packages/typespec-go/src/emitter.ts

```diff
@@ -24,14 +24,88 @@ import { generateMetadataFile } from '../../codegen.go/src/metadata.js';
 import { CodeModelError } from '../../codemodel.go/src/errors.js';
 import { existsSync } from 'fs';
 import { mkdir, readFile, writeFile } from 'fs/promises';
-import { EmitContext, NoTarget } from '@typespec/compiler';
+import { DiagnosticSeverity, EmitContext, NoTarget } from '@typespec/compiler';
 import 'source-map-support/register.js';
 import { reportDiagnostic } from './lib.js';
 import { CodegenError } from '../../codegen.go/src/errors.js';
+import { execSync } from 'child_process';
 
 export async function $onEmit(context: EmitContext<GoEmitterOptions>) {
   try {
     await generate(context);
+
+    const afterGenerateFileExists = existsSync(`${context.emitterOutputDir}/after_generate.go`);
```

Comment:
Rather than trigger based on a file. Can we use the tspconfig.yaml?
i.e.
From this file, https://github.com/Azure/azure-rest-api-specs/blob/f2acb480cbbd549eb2aafff1372dbddf0f8da472/specification/eventgrid/Azure.Messaging.EventGrid.SystemEvents/tspconfig.yaml, We'd add a `post-generate-step` option?
```
  "@azure-tools/typespec-go":
    module: "github.com/Azure/azure-sdk-for-go/{service-dir}/{package-dir}"
    service-dir: "sdk/messaging/eventgrid"
    package-dir: "azsystemevents"
    module-version: "0.0.1"
    emitter-output-dir: "{project-root}"
    slice-elements-byval: true
    flavor: "azure"
```

Reply:
That does have the side-effect of making it explicit which IMO is good. However, there is a one-time cost to go and update the necessary `tspconfig.yaml` files with this info.